pkg/image/importer/dockerv1client/scheme.go

```diff
@@ -0,0 +1,68 @@
+package dockerv1internalclient
+
+import (
+	docker "github.com/fsouza/go-dockerclient"
+	imagev1 "github.com/openshift/api/image/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+var (
+	DockerClientScheme = runtime.NewScheme()
+)
+
+// this is the only entrypoint which deals in github.com/fsouza/go-dockerclient.Image and expects to use our conversion capability to coerce an external
+// type into an api type.  Localize the crazy here.
+// TODO this scheme needs to be configurable or we're going to end up with weird problems.
+func init() {
+	utilruntime.Must(imagev1.Install(DockerClientScheme))
+	utilruntime.Must(imageapi.Install(DockerClientScheme))
+
+	err := DockerClientScheme.AddConversionFuncs(
+		// Convert docker client object to internal object
+		func(in *docker.Image, out *imageapi.DockerImage, s conversion.Scope) error {
```

Comment:
this is a move?

Reply:
schema was extracted from the client.go to make it more visible and reworked to not poison the legacyscheme.